sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -987,6 +1051,64 @@ public virtual Response<AlertingHook> CreateHook(AlertingHook hook, Cancellation
             }
         }
 
+        /// <summary>
+        /// </summary>
+        public virtual async Task<Response> UpdateHookAsync(string hookId, AlertingHook hook, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(hook, nameof(hook));
+
+            Guid hookGuid = ClientCommon.ValidateGuid(hookId, nameof(hookId));
```

Comment:
We've been adding null-empty-checks for ids in the service method body, out of the `ValidateGuid` method. I'm okay with changing the approach, but I wanted us to be consistent.